### PR TITLE
[enhancement] proxy: add support for stmt prepare cmd.

### PR DESCRIPTION
**Pull request by volgariver6**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [x] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #13778

## What this PR does / why we need it:
add support for stmt prepare cmd.

### CODE REVIEW

**Comment by matrix-meow** [comment]
@volgariver6 Thanks for your contributions!

Review:

Title: [enhancement] proxy: add support for stmt prepare cmd.

Body: The pull request is an enhancement that adds support for the stmt prepare command in the proxy.

Changes:
1. Added a new constant `cmdStmtPrepare` in `mysql_conn_buf.go` to represent the stmt prepare command.
2. Modified the `makeEvent` function in `event.go` to check for the stmt prepare command and create a `PrepareEvent` if found.
3. Added a new function `isCmdStmtPrepare` in `util.go` to check if a given byte array represents the stmt prepare command.
4. Added a new test case `TestIsCmdStmtPrepare` in `util_test.go` to test the `isCmdStmtPrepare` function.

Overall, the changes made in this pull request seem to be fine. However, there are a few suggestions for improvement and potential security issues to address:

1. Security Issue: The `isCmdStmtPrepare` function does not perform any validation on the byte array to ensure that it represents a valid stmt prepare command. This can potentially lead to security vulnerabilities if the input is manipulated. It is recommended to add additional validation to ensure the byte array is in the expected format.

2. Optimization: The `isCmdStmtPrepare` function can be optimized by checking the length of the byte array before checking the specific byte value. This can help improve performance by avoiding unnecessary checks.

3. Test Coverage: It would be beneficial to add more test cases to cover different scenarios for the `isCmdStmtPrepare` function, such as testing with invalid byte arrays or edge cases.

To address these suggestions, the following changes can be made:

1. Add validation to the `isCmdStmtPrepare` function to ensure the byte array is in the expected format. This can be done by checking the length of the byte array and verifying that the specific byte value matches the stmt prepare command.

Example:
```go
func isCmdStmtPrepare(p []byte) bool {
	if len(p) > 4 && p[4] == byte(cmdStmtPrepare) {
		// Additional validation to ensure the byte array is in the expected format
		// Add code here to check other bytes if necessary
		return true
	}
	return false
}
```

2. Optimize the `isCmdStmtPrepare` function by checking the length of the byte array before checking the specific byte value. This can be done by moving the length check before the byte value check.

Example:
```go
func isCmdStmtPrepare(p []byte) bool {
	if len(p) > 4 && p[4] == byte(cmdStmtPrepare) {
		// Additional validation to ensure the byte array is in the expected format
		// Add code here to check other bytes if necessary
		return true
	}
	return false
}
```

3. Add more test cases to cover different scenarios for the `isCmdStmtPrepare` function. This can include testing with invalid byte arrays, edge cases, and different lengths.

Example:
```go
func TestIsCmdStmtPrepare(t *testing.T) {
	var data []byte
	ret := isCmdStmtPrepare(data)
	require.False(t, ret)

	data = []byte{0, 0, 0, 0, 20, 0}
	ret = isCmdStmtPrepare(data)
	require.False(t, ret)

	data = []byte{0, 0, 0, 0, byte(cmdStmtPrepare), 0}
	ret = isCmdStmtPrepare(data)
	require.True(t, ret)

	// Add more test cases here
}
```

By addressing these suggestions, the codebase will be more secure and optimized, and the test coverage will be improved.

**Review by zhangxu19830126 — APPROVED** [review]

File: pkg/proxy/event.go [hunk 1]
```diff
@@ -109,6 +109,8 @@ func makeEvent(msg []byte) (IEvent, bool) {
 		}
 	} else if isCmdInitDB(msg) {
 		return makeInitDBEvent(getStatement(msg)), false
+	} else if isCmdStmtPrepare(msg) {
+		return makePrepareEvent(getStatement(msg)), false
```
Comment (volgariver6) [inline pkg/proxy/event.go:113]: still need to change the sql stmt
